### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by Zincion8**
This was painful

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.78. However, no evidence supports this result.

**Model and functions:**

- Student try to use PyTorch to construct the 2D-UNet models. However, the constructed model is one simplified variant of UNet.
- No data augmentation.
- No validation application to supervise the model training. Students are expected to use the training dataset to train the model and the validation dataset to observe the model performance in the training phase, then to use the test dataset to test the model performance on the unseen data.
-  Neither dice similarity coefficient nor dice loss definition in the code.  

**Code design:** Incomplete code design.
- Dice loss are expected to be applied in the segmentation task but didn't appear in the project code.

**Code comment and docstring:**

- Good code comments 
- Minimal function docstrings

**Difficulty:** Easy.

**Additional Comments:**

- intermediate numbers of commits
- Figure link error in the report leading to image disappearance.
- The Dice scores in the report cannot be generated from the project code as there is no dice score definition function in the code.
-  The last line in the "How It Works" section, "PLEASE ADD PATHS IN FOR YOU", indicates that parts or the entire report is suspected generated by the ChatGPT.
- An experiment analysis and conclusion are expected to be in the report.
-  Wrong reference format: reference list should be organized together at the end of the report.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Adequate design and implementation. </td><td>-1</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. No loss defined, Model is a varient of Unet.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Minimal doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Some markup errors. Incomplete.</td><td>-4</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Wrong reference format: reference list should be organized together at the end of the report.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-22</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Zeke Johnson

**Comment by shakes76** [comment]
Not correct UNet -2 additionally.
